lib/backends/kv-backend.test.js

```diff
@@ -446,5 +485,38 @@ describe('kv-backend', () => {
         jsonProperty: 'eyJzb21lS2V5Ijp7Im15VGV4dCI6InRleHQifX0=' // base 64 value of: {"someKey":{"myText":"text"}}
       })
     })
+
+    it('handles different types of binary data returned by backends', async () => {
+      kvBackend._get.onCall(0).resolves(Buffer.from('YmluYXJ5Cg==', 'base64')) // base64 of binary as a Buffer
+      kvBackend._get.onCall(1).resolves(Buffer.from('YmluYXJ5Cg==', 'base64')) // base64 of binary as a Buffer
+      kvBackend._get.onCall(2).resolves('YmluYXJ5Cg==') // base64 of binary as String
+      // e.g. AWS Secrets Manager will return `SecretBinary` as a Buffer and `SecretString` as a String
+
+      const manifestData = await kvBackend.getSecretManifestData({
+        spec: {
+          data: [{
+            key: 'binaryPropertyKey1',
+            name: 'binaryPropertyName1',
+            isBinary: true
+          }, {
+            key: 'binaryPropertyKey2',
+            name: 'binaryPropertyName2'
+            // isBinary: false, but will have no impact if the backend returns a Buffer
+          }, {
+            key: 'stringPropertyKey3',
+            name: 'stringPropertyName3'
+            // isBinary: false,
+            // must be set to true to ensure base64-encoded string in the backend
+            // is not encoded in base64 again
+          }]
+        }
+      })
+
+      expect(manifestData).deep.equals({
+        binaryPropertyName1: 'YmluYXJ5Cg==', // base64 of binary (unchanged)
+        binaryPropertyName2: 'YmluYXJ5Cg==', // base64 of binary (unchanged)
+        stringPropertyName3: 'WW1sdVlYSjVDZz09' // base64 of base64 of binary
+      })
+    })
```

Comment:
Can we add a test variant which explicitly sets `isBinary: false` where the value is a string?

Reply:
@Flydiverny Makes sense, done!